### PR TITLE
MBS-12566: Fix incorrect URL shortener detection

### PR DESCRIPTION
**Pull request by mwiencek**
https://tickets.metabrainz.org/browse/MBS-12566

As the ticket explains, our RegExps should escape their dots (t.co
should not match taco), and also match the hostname correctly (t.co
should not match t.co.example).

### CODE REVIEW

File: root/static/scripts/tests/index-web.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ require('./GuessCase');
 require('./guessFeat');
 require('./i18n');
 require('./i18n/expand2');
-require('./relationship-editor');
+//require('./relationship-editor');
```
Comment (reosarevok) [inline root/static/scripts/tests/index-web.js:14]: This seems unrelated from the other changes? (it also causes an eslint warning)

**Review by reosarevok — CHANGES_REQUESTED** [review]
One issue, but other than that this seems like a big improvement, thanks! Every time I added a new shortener I thought "we should add a test for this" but got lazy, that's what I get! (surprising this bug wasn't reported a lot sooner).

File: root/static/scripts/tests/index-web.js (hunk 1 above)
Comment (mwiencek) [reply root/static/scripts/tests/index-web.js]: Oops, I had commented out that test locally because it was failing for some reason, then accidentally committed it.
